### PR TITLE
Extend PP with ppRename: nicer compose of WorkspaceNames, ClickableWorkspaces, …

### PR DESCRIPTION
**Pull request by liskin**
#### [X.H.DynamicLog: Add ppRename (composable tag augmentation)](../commit/91010f6eb916aedeb8f9f8b5d0f456a688c5f094)

This one is a Reader in WindowSpace, and therefore significantly
simplifies the composition of WorkspaceNames, IndependentScreens,
ClickableWorkspaces and possibly other similar modules.

Related: https://github.com/xmonad/xmonad-contrib/pull/390
Related: https://github.com/xmonad/xmonad-contrib/pull/462

#### [X.A.WorkspaceNames, X.L.IndependentScreens, X.U.ClickableWorkspaces: Use ppRename](../commit/275af4d4df2c20a70d238313e17960938e8bf6ee)

Also, drop now useless integrations from X.U.ClickableWorkspaces:
workspaceNamesPP, marshallPP and clickablePP can now be composed
directly using >>= and it just works.

Related: https://github.com/xmonad/xmonad-contrib/pull/390
Related: https://github.com/xmonad/xmonad-contrib/pull/462

### Checklist

  - [X] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [X] I updated the `CHANGES.md` file

  - n/a I updated the `XMonad.Doc.Extending` file (if appropriate)

### CODE REVIEW

**Review by slotThe — COMMENTED** [review]
I really like this! :+1:   
 
I played around a bit with making `ppRename` a `String -> Reader WindowSpace String` (so we didn't have to make that assumption in a comment (yes I know it's all isomorphic, but sometimes names _do_ matter :))) but I feel like the ergonomics just aren't there

File: XMonad/Layout/IndependentScreens.hs [hunk 1]
```diff
@@ -135,15 +135,8 @@ countScreens = fmap genericLength . liftIO $ openDisplay "" >>= liftA2 (<*) getS
...
-marshallPP s pp = pp {
-    ppCurrent           = ppCurrent         pp . unmarshallW,
-    ppVisible           = ppVisible         pp . unmarshallW,
-    ppHidden            = ppHidden          pp . unmarshallW,
-    ppHiddenNoWindows   = ppHiddenNoWindows pp . unmarshallW,
-    ppVisibleNoWindows  = ppVisibleNoWindows pp <&> (. unmarshallW),
-    ppUrgent            = ppUrgent          pp . unmarshallW,
-    ppSort              = fmap (marshallSort s) (ppSort pp)
-    }
+marshallPP s pp = pp { ppRename = ppRename pp <=< pure . unmarshallW
```
[the remark this answers is not in the document]
Comment (liskin) [reply XMonad/Layout/IndependentScreens.hs:138]: It just didn't occur to me that I can use `.` here :-)

**Comment by liskin** [comment]
Oh, I thought we'd wait for #465 to not make it difficult to rebase. If we're merging DynamicLog stuff we can merge #471 as well and then rebase #465.

**Comment by slotThe** [comment]
On Sat, Mar 20 2021 08:39, Tomáš Janoušek wrote:
> Oh, I thought we'd wait for #465 to not make it difficult to
> rebase. If we're merging DynamicLog stuff we can merge #471 as well
> and then rebase #465.

Whoops.  Sorry, must have missed that; I hope rebasing will not be too
annoying now.

**Pull request merged**